src/wallet/wallet.cpp

```diff
@@ -1139,6 +1163,82 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlockIndex *pindex,
     }
 }
 
+void CWallet::UpdatedBlockHeaderTip(bool fInitialDownload, const CBlockIndex *pindexNew)
+{
+    int bestSPVBlockHeight = 0;
+    {
+        LOCK2(cs_main, cs_wallet);
+        bestSPVBlockHeight = nBestSpvHeight;
+    }
+
+    // check if the new tip extends the wallets current known best headers tip
+    const CBlockIndex *pCurrent = pindexNew;
```

Comment:
Can pindexNew ever be null? If if it is null, should fReOrg always be true in this case, or should fReorg only be true if bestSpvBlockHash is null? Comment might be useful to clarify.

If pIndexNew should never be null, would suggest either asserting this or changing the while to a do-while, and removing the "pindexNew && ..." check below.